pkg/instancetype/instancetype.go

```diff
@@ -522,17 +523,56 @@ func (m *methods) findClusterInstancetype(vm *virtv1.VirtualMachine) (*instancet
 	return instancetype, nil
 }
 
+func resolveConflicts(field *k8sfield.Path, requestType k8sv1.ResourceName, vmiSpec *virtv1.VirtualMachineInstanceSpec, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) (Conflicts, bool) {
+	hasGuestField := false
+
+	switch requestType {
+	case k8sv1.ResourceCPU:
+		if vmiSpec.Domain.CPU != nil {
+			return Conflicts{field.Child("domain", "cpu")}, true
+		}
+		if instancetypeSpec.CPU.Guest != 0 {
+			hasGuestField = true
+		}
+	case k8sv1.ResourceMemory:
+		if vmiSpec.Domain.Memory != nil {
+			return Conflicts{field.Child("domain", "memory")}, true
+		}
+		if !instancetypeSpec.Memory.Guest.IsZero() {
+			hasGuestField = true
+		}
+	}
+
+	if _, hasRequests := instancetypeSpec.Resources.Requests[requestType]; hasRequests && hasGuestField {
+		return Conflicts{field.Child("domain", "resources", "requests", string(requestType))}, true
```

Comment:
I also wonder if we should append the following list of conflicts together to make the error a little more descriptive to the end user if they have already set requests and limits etc.